### DISCUSSION TITLE
Selective include/exclude the packages that mapsctruct should scan for processing

### DISCUSSION THREAD

**Discussion by rahulagrawal** [mapstruct/mapstruct#3494] [Q&A]
We have a kotlin springboot project with many 3rd party libraries.
We are using mapstruct with KAPT using the following library.
org.mapstruct:mapstruct-processor:1.5.5.Final

Currently, it seems that everything under the roof - ie our own code as well as 3rpd party jars are scanned my mapstruct for transformation.
We are looking for ways to restrict the scanning to be done for only certain packages inside our source code.
May I know if there is a way to do this in mapstruct or this is something that KAPT should be providing.

**Comment by filiphr** [comment]
@rahulagrawal I am not sure that I understand what you are asking? I am not sure how exactly KAPT works. However, with the Java annotation processing we are going to get all the classes that are being compiled (through the java compiler) and we check if they are annotated with `@Mapper`. We don't perform any scanning, the scanning should be done by KAPT.

**Reply by rahulagrawal** [reply]
Thanks a lot for helping me. 
Let me attempt to explain what I am trying to understand and explore.

We have our own kotlin code written by our team in which we annotate some classes with @Mapper. 
And we are using KAPT - https://kotlinlang.org/docs/kapt.html to get annotation processing in kotlin.
We also use 3rd party libraries.

So when you mention that  - ```_However, with the Java annotation processing we are going to get all the classes that are being compiled (through the java compiler_)```
Does it mean that only the source code that we have written and not the 3rd party libraries would be checked if they are annotated with @Mapper ? And if that is the case , then it seems that I need to look deeper into KAPT to understand why it seems to going over everything and not just our  code. 

Thanks and Regards,

**Reply by filiphr** [reply]
>Does it mean that only the source code that we have written and not the 3rd party libraries would be checked if they are annotated with `@Mapper`?

Indeed, in theory KAPT should only pass the source code from the module that you have written to the annotation processor. if KAPT is passing the 3rd party libraries then MapStruct will try to generate code for them. However, we have no way of knowing whether the classes are coming from 3rd party libraries or from the code you've written.

For example, when using Java and Maven. If you have multiple modules the `maven-compiler-plugin` is going to invoke the javac compiler for each module separately and it is going to pass the source of each module to the compiler. KAPT (or the plugin you use to invoke KAPT) should be doing the same.
